src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.cs

```diff
@@ -0,0 +1,2565 @@
+						popup.HorizontalOffset = UntargetedTipNearPlacementOffset((float)windowBoundsInCoreWindowSpace.X, offset.Left);
+						break;
+
+					case TeachingTipPlacementMode.RightTop:
+						popup.VerticalOffset = UntargetedTipNearPlacementOffset((float)windowBoundsInCoreWindowSpace.Y, offset.Top);
+						popup.HorizontalOffset = UntargetedTipFarPlacementOffset((float)windowBoundsInCoreWindowSpace.Width, finalTipWidth, offset.Right);
+						break;
+
+					case TeachingTipPlacementMode.RightBottom:
+						popup.VerticalOffset = UntargetedTipFarPlacementOffset((float)windowBoundsInCoreWindowSpace.Height, finalTipHeight, offset.Bottom);
+						popup.HorizontalOffset = UntargetedTipFarPlacementOffset((float)windowBoundsInCoreWindowSpace.Width, finalTipWidth, offset.Right);
+						break;
+
+					case TeachingTipPlacementMode.Center:
+						popup.VerticalOffset = UntargetedTipCenterPlacementOffset((float)windowBoundsInCoreWindowSpace.Y, (float)windowBoundsInCoreWindowSpace.Height, finalTipHeight, offset.Top, offset.Bottom);
+						popup.HorizontalOffset = UntargetedTipCenterPlacementOffset((float)windowBoundsInCoreWindowSpace.X, (float)windowBoundsInCoreWindowSpace.Width, finalTipWidth, offset.Left, offset.Right);
+						break;
+
+					default:
+						throw new InvalidOperationException("Provided placement is not supported");
+				}
+			}
+
+			return tipDoesNotFit;
+		}
+
+		private void UpdateSizeBasedTemplateSettings()
+		{
+			var templateSettings = TemplateSettings;
+			(double width, double height) GetActualSize()
+			{
+				var contentRootGrid = m_contentRootGrid;
+				if (contentRootGrid != null)
+				{
+					var width = contentRootGrid.ActualWidth;
+					var height = contentRootGrid.ActualHeight;
+					return (width, height);
+				}
+				return (0.0, 0.0);
+			}
+			var (width, height) = GetActualSize();
+
+			switch (m_currentEffectiveTailPlacementMode)
+			{
+				case TeachingTipPlacementMode.Top:
+					templateSettings.TopRightHighlightMargin = OtherPlacementTopRightHighlightMargin(width, height);
+					templateSettings.TopLeftHighlightMargin = TopEdgePlacementTopLeftHighlightMargin(width, height);
+					break;
+				case TeachingTipPlacementMode.Bottom:
+					templateSettings.TopRightHighlightMargin = BottomPlacementTopRightHighlightMargin(width, height);
+					templateSettings.TopLeftHighlightMargin = BottomPlacementTopLeftHighlightMargin(width, height);
+					break;
+				case TeachingTipPlacementMode.Left:
+					templateSettings.TopRightHighlightMargin = OtherPlacementTopRightHighlightMargin(width, height);
+					templateSettings.TopLeftHighlightMargin = LeftEdgePlacementTopLeftHighlightMargin(width, height);
+					break;
+				case TeachingTipPlacementMode.Right:
+					templateSettings.TopRightHighlightMargin = OtherPlacementTopRightHighlightMargin(width, height);
+					templateSettings.TopLeftHighlightMargin = RightEdgePlacementTopLeftHighlightMargin(width, height);
+					break;
+				case TeachingTipPlacementMode.TopLeft:
```

Comment:
Codacy found an issue: [Either merge this case with the identical one on line 694 or change one of the implementations.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)